integrations/lib/multiwoven/integrations/destination/http/client.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+require "base64"
+module Multiwoven
+  module Integrations
+    module Destination
+      module Http
+        include Multiwoven::Integrations::Core
+        class Client < DestinationConnector
+          MAX_CHUNK_SIZE = 10
+          def check_connection(connection_config)
+            connection_config = connection_config.with_indifferent_access
+            destination_url = connection_config[:destination_url]
+            request = Multiwoven::Integrations::Core::HttpClient.request(
+              destination_url,
+              HTTP_OPTIONS
+            )
+            if success?(request)
+              success_status
+            else
+              failure_status(nil)
+            end
+          rescue StandardError => e
+            handle_exception("HTTP:CHECK_CONNECTION:EXCEPTION", "error", e)
+            failure_status(e)
+          end
+
+          def discover(_connection_config = nil)
+            catalog_json = read_json(CATALOG_SPEC_PATH)
+            catalog = build_catalog(catalog_json)
+            catalog.to_multiwoven_message
+          rescue StandardError => e
+            handle_exception(
+              "HTTP:DISCOVER:EXCEPTION",
+              "error",
+              e
+            )
+          end
+
+          def write(sync_config, records, _action = "create")
+            connection_config = sync_config.destination.connection_specification.with_indifferent_access
+
+            url = connection_config[:destination_url]
+            username = connection_config[:username]
+            password = connection_config[:password]
+
+            auth_key = convert_to_auth_key(username, password) if username && password
+
+            write_success = 0
+            write_failure = 0
+            records.each_slice(MAX_CHUNK_SIZE) do |chunk|
+              payload = create_payload(chunk)
+              response = Multiwoven::Integrations::Core::HttpClient.request(
+                url,
+                sync_config.stream.request_method,
+                payload: payload,
+                headers: auth_headers(auth_key)
```

Comment:
@AnthonyR-AI2  Auth headers needs to designed in a generic fashion. This is not extendable to different URL endpoints: What if the endpoint except` algolia-api-key` as header for the `auth_token`?

Reply:
These have been changed to generic headers.